### PR TITLE
gen-420 and some type fixes

### PR DESCRIPTION
**Pull request by littleredcomputer**
cleans up a few type errors and adds a missing else case

### CODE REVIEW

File: src/genjax/_src/core/generative/choice_map.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ def select_none():
 
 @Pytree.dataclass
 class StaticSel(Selection):
-    addr: StaticAddressComponent = Pytree.static()
+    addr: StaticAddressComponent | EllipsisType = Pytree.static()
```
Comment (femtomc) [inline src/genjax/_src/core/generative/choice_map.py]: Change to `ExtendedStaticAddressComponent` for consistency

**Review by femtomc — CHANGES_REQUESTED** [review]
Single LoC change otherwise looks great!

File: src/genjax/_src/inference/smc.py [hunk 2]
```diff
@@ -293,7 +293,7 @@ class ImportanceK(SMCAlgorithm):
     importance sampling."""
 
     target: Target
-    q: Optional[SampleDistribution] = Pytree.field(default=Pytree.const(None))
+    q: SampleDistribution = Pytree.field(default=Pytree.const(None))
```
Comment (femtomc) [inline src/genjax/_src/inference/smc.py]: `None` is officially allowed in JAX, so I think we should restore the `Optional`, and just have `Pytree.field(default=None)`

Comment (littleredcomputer) [reply src/genjax/_src/inference/smc.py]: OK. (To make that work, I have changed the definition of `static_check_none`). If you don't think anyone would ever use `Const(None)` then we can remove that method and just check `is None` at the call sites

**Comment by littleredcomputer** [comment]
I also fixed a problem where smc was defining the instance variable `target`, which was shadowing the method `GenerativeFunction.target()` in an incompatible way (method vs. member). Since nobody was using the method and it was undocumented I deleted it.

**Comment by littleredcomputer** [comment]
@femtomc ptal

**Review by femtomc — APPROVED** [review]

**Pull request merged**